clients/ts/signalr/src/HubConnectionBuilder.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import { HttpConnection, IHttpConnectionOptions } from "./HttpConnection";
+import { HubConnection, JsonHubProtocol } from "./HubConnection";
+import { IHubProtocol } from "./IHubProtocol";
+import { ILogger, LogLevel } from "./ILogger";
+import { HttpTransportType } from "./ITransport";
+import { NullLogger } from "./Loggers";
+import { Arg, ConsoleLogger } from "./Utils";
+
+export class HubConnectionBuilder {
+    /** @internal */
+    public protocol: any;
```

Comment:
We need to talk a bit about this. It's really the only way to test the `transportType` overload of `withUrl`.

In our tsconfig, we set `stripInternal: true`, which tells TypeScript to **strip** items marked with the JSDoc `@internal` annotation **from our .d.ts files**. This means:

* The properties are still "public" in JavaScript, meaning they can be read/written from JavaScript
* TypeScript code that references our compiled output (which are `.js` files with `.d.ts` buddies) cannot access this property (unless they cast to `any`, which "turns off" type checking)
* TypeScript code that references the `.ts` file directly (our tests) can see the property just fine.

This is basically The Only Way to do something like "Internals Visible To" in TypeScript.